content/docs/04-ai-sdk-ui/04-generative-ui.mdx

```diff
@@ -0,0 +1,347 @@
+---
+title: Generative UI
+description: Learn how to build Generative UI with AI SDK UI.
+---
+
+# Generative UI
+
+Generative UI is the process of allowing a large language model (LLM) to go beyond text and "generate UI". This creates a more engaging and AI-native experience for users.
+
+<WeatherSearch />
+
+At the core of generative UI are [ tools ](/docs/ai-sdk-core/tools-and-tool-calling), which are like programs you provide to the model to perform specialized tasks like getting the weather in a location. The model can decide when and how to use these tools based on the context of the conversation.
+
+Generative UI is simply the process of piping the results of a tool call to a React component. Here's how the process works:
```

Comment:
cut `simply`

Reply:
`Here's how it works:`

Reply:
I would rephrase to avoid the word "piping" since it's not immediately obvious what it means